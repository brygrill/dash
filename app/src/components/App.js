import React, { Component } from 'react';

import ErrorBoundary from './ErrorBoundary';
import Router from './router/Router';

export default class App extends Component {
  render() {
    return (
      <ErrorBoundary>
        <Router />
      </ErrorBoundary>
    );
  }
}
