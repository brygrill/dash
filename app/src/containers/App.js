/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { Match, BrowserRouter, Miss } from 'react-router';

import Home from '../components/Home';
import Dashboard from './Dashboard';
import NotFound from '../components/NotFound';

class App extends Component {

  render() {
    return (
      <BrowserRouter>
        <div>
          <h1>App</h1>
          <Match exactly pattern="/" component={Home} />
          <Match pattern="/dashboard" component={Dashboard} />
          <Miss component={NotFound} />
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
