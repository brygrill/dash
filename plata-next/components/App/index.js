import React, { PropTypes } from 'react';
import { Container } from 'semantic-ui-react';
import Head from '../Head';

const App = (props) => {
  return (
    <div>
      <Head title={props.pgTitle} />
      <Container fluid className="plata-height-100">
        <div className="plata-height-100">
          {props.children}
        </div>
      </Container>
    </div>
  );
};

App.propTypes = {
  children: PropTypes.node,
  pgTitle: PropTypes.string,
};

export default App;
