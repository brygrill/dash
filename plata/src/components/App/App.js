/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import Match from 'react-router/Match';
import Miss from 'react-router/Miss';
import { Container, Header } from 'semantic-ui-react';
import { firebaseAuth } from '../../firebase';
import Home from '../Home/Home';
import Login from '../Login/Login';
import './App.css';

const NoMatch = () => (
  <h1>404!</h1>
);

class App extends Component {
  componentDidMount() {
    firebaseAuth.onAuthStateChanged((user) => {
      console.log(user);
    });
  }
  render() {
    return (
      <div className="App">
        <Container>
          <Header as="h1">Plata App!!</Header>
          <Match exactly pattern="/" component={Home} />
          <Match pattern="/login" component={Login} />
          <Miss component={NoMatch} />
        </Container>
      </div>
    );
  }
}

export default App;
