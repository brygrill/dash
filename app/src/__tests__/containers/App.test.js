import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../../containers/App';

it('App container renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('App shows the home page!', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
