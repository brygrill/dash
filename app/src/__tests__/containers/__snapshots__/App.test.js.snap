exports[`test App shows the home page! 1`] = `
<div>
  <h1>
    App
  </h1>
  <h1>
    Home!
  </h1>
</div>
`;

exports[`test Home pages shows the home page! 1`] = `
<div>
  <h1>
    App
  </h1>
  <h1>
    Home!
  </h1>
</div>
`;
